yspryn/hw10/recreate_generators.py

```diff
@@ -0,0 +1,68 @@
+def my_zip(*iterable):
+
+    position = 0
+    min_length = min(len(item) for item in iterable)
+
+    while position < min_length:
+        res = []
+        for i in range(len(iterable)):
+            res.append(iterable[i][position])
+        yield tuple(res)
+        position += 1
```

Comment:
in this case you can expect that you can use  iter for iterables,

so better if you rely on  iter next interface,


since iterables can be - collections iterators generators

Reply:
same for others in this module